rotkehlchen/db/upgrades/v39_v40.py

```diff
@@ -83,6 +128,114 @@ def _add_base_chain_data(write_cursor: 'DBCursor') -> None:
+                    log.error(f'Skipped a ledger action from migration to history event due {e}')  # noqa: E501
+                    continue
+
+                usd_value = amount * rate
+
+            ledger_action_type = entry[1]
+            notes = f'Migrated from a ledger action of {LEDGER_ACTION_TYPE_TO_NAME[ledger_action_type]} type'  # due to DB consistency there should be no key errors # noqa: E501
+            if entry[8] is not None:
+                notes = f'{entry[8]}. {notes}'
+            if entry[7] is not None:
+                notes = f'{notes}. {entry[7]}'
+
+            # map the types. Making sure to have the as default/valid combinations
+            if ledger_action_type in ('A', 'D', 'G', 'H'):  # income, dividends income, gift, grant action type  # noqa: E501
+                event_type = 'receive'
+                event_subtype = 'none'
+            elif ledger_action_type in ('B', 'C'):  # expense, loss action type
+                event_type = 'spend'
+                event_subtype = 'none'
+            elif ledger_action_type == 'E':
+                event_type = 'receive'
+                event_subtype = 'donate'
+            elif ledger_action_type == 'F':
+                event_type = 'receive'
+                event_subtype = 'airdrop'
+            else:
+                raise AssertionError('Should never happen. Guaranteed by DB consistency')
+
+            history_events.append((
+                1,  # entry_type -> HISTORY_EVENT
+                f'{MIGRATION_PREFIX}{uuid4().hex}',  # event_identifier
+                0,  # sequence_index
+                entry[0] * 1000,  # timestamp
+                location,
+                None,  # location_label
+                entry[4],  # asset
+                entry[3],  # amount
+                str(usd_value),  # usd_value
+                notes,
+                event_type,
+                event_subtype,
+            ))
+
+    write_cursor.executemany("""INSERT OR IGNORE INTO history_events(
+    entry_type,
+    event_identifier,
+    sequence_index,
+    timestamp,
+    location,
+    location_label,
+    asset,
+    amount,
+    usd_value,
+    notes,
+    type,
+    subtype) VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)""", history_events)
+    write_cursor.execute('DROP TABLE IF EXISTS ledger_actions;')
+    write_cursor.execute('DROP TABLE IF EXISTS ledger_action_type;')
+    write_cursor.execute('DELETE from settings WHERE name=?', ('taxable_ledger_actions',))
+
+    with conn.read_ctx() as cursor:  # migrate coinbase ledger action query ranges
+        cursor.execute("SELECT name FROM used_query_ranges WHERE name LIKE '%_ledger_actions_%'")
+        for entry in cursor:
+            name = entry[0]
+            try:
+                write_cursor.execute(
+                    'UPDATE used_query_ranges SET name = ? WHERE name = ?',
+                    (name.replace('_ledger_actions_', '_history_events_'), name),
+                )
```

Comment:
is it worth the extra variable storage just for the case of the query raising the exception?

Reply:
I don't understand the question

Reply:
so you use `name = entry[0]` to reuse it in the error message but then you have one extra variable creation just for reusing it in the except log mesasge. Why not use entry[0] in both places?

Reply:
readability and size of the code I guess. This code changed a lot as I was working on the PR

Reply:
If it makes you happy I can remove it also